src/backend/drm/compositor/mod.rs

```diff
@@ -1671,19 +1680,9 @@ where
             }),
         };
 
-        // test that we can scan-out the primary plane
-        //
-        // Note: this should only fail if the device has been
-        // deactivated or we lost access (like during a vt switch)
-        next_frame_state
-            .test_state(
-                &self.surface,
-                self.supports_fencing,
-                self.planes.primary.handle,
-                primary_plane_state,
-                self.surface.commit_pending(),
-            )
-            .map_err(FrameError::DrmError)?;
+        // unconditionally set the primary plane state
+        // if this would fail the test we are screwed anyway
+        next_frame_state.set_state(self.planes.primary.handle, primary_plane_state);
```

Comment:
Yeah, this only sets the "initial" frame state to a swapchain slot. The only real difference is that this now no longer fails in case the device has been deactivated (tty switch), so upstream would notice that a bit later. But this is no issue imo as this can happen anytime anyway.